### PR TITLE
Russian translation (again) & little visual improvement

### PR DESCRIPTION
**Pull request by tltneon**
I replaced original Prototype font with new one with supporting Cyrillic symbols (for russian, ukrainian, bulgarian, serbian an so on)

image

And also edit CSS rules that sets name exactly at the middle of screen.

Next one. I made little feature: when card has a huge of description, it can be scrolled.
It will look like (pointed scroll at the right side)
image image

Needs almost for every locale
image image
image image

### CODE REVIEW

**Comment by tltneon** [comment]
Checks fails but it successfully builds on my machine

```bash
C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars>npm run build   && cmd /k npm run start

> terraforming-mars@1.0.0 build C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> npm run make:static && npm run build:server && npm run build:client


> terraforming-mars@1.0.0 make:static C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> npm run make:css && npm run make:json


> terraforming-mars@1.0.0 make:css C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> lessc src/styles/common.less build/styles.css && node src/server/tools/gzip.js


> terraforming-mars@1.0.0 make:json C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> node make_static_json.js


> terraforming-mars@1.0.0 build:server C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> tsc --build src/tsconfig.json


> terraforming-mars@1.0.0 build:client C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> npm run make:cards && cross-env NODE_ENV=production webpack


> terraforming-mars@1.0.0 make:cards C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> node build/src/server/tools/export_card_rendering.js

asset main.js 1.54 MiB [compared for emit] [minimized] [big] (name: main) 4 related assets
orphan modules 462 KiB [orphan] 350 modules
runtime modules 1.13 KiB 5 modules
cacheable modules 2.36 MiB
  modules by path ./src/client/components/ 783 KiB 243 modules
  modules by path ./src/common/ 94.2 KiB 40 modules
  modules by path ./src/client/utils/*.ts 12.9 KiB 6 modules
  json modules 760 KiB 5 modules
  modules by path ./node_modules/ 747 KiB 4 modules
  modules by path ./src/client/*.ts 2.87 KiB 2 modules
  modules by path ./src/client/directives/*.ts 4.7 KiB
    ./src/client/directives/TrimWhitespace.ts 439 bytes [built] [code generated]
    ./src/client/directives/i18n.ts 4.28 KiB [built] [code generated]
  modules by path ./src/client/cards/*.ts 1.53 KiB
    ./src/client/cards/ClientCardManifest.ts 1.13 KiB [built] [code generated]
    ./src/client/cards/SpecialTags.ts 409 bytes [built] [code generated]

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  main.js (1.54 MiB)
  main.js.gz (270 KiB)
  main.js.map.gz (625 KiB)
  main.js.map.br (516 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  main (1.54 MiB)
      main.js


WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/

webpack 5.39.0 compiled with 3 warnings in 40795 ms

> terraforming-mars@1.0.0 start C:\Users\user\Desktop\terraforming-mars-main\terraforming-mars
> node build/src/server/server.js

[2023-05-01 00:05:18 GMT+0400] Connecting to SQLite database.
[2023-05-01 00:05:18 GMT+0400] {
  "type": "SQLite",
  "path": "C:\\Users\\user\\Desktop\\terraforming-mars-main\\terraforming-mars\\db\\game.db",
  "size_bytes": 737280
}
[2023-05-01 00:05:18 GMT+0400] Starting cdd9decee, built at Sat, 29 Apr 2023 00:16:33 -0400
[2023-05-01 00:05:18 GMT+0400] Starting server on port 8080
[2023-05-01 00:05:18 GMT+0400] The secret serverId for this server is f498620f4da.
[2023-05-01 00:05:18 GMT+0400] Adminsitrative routes can be found at admin?serverId=f498620f4da
[2023-05-01 00:05:18 GMT+0400] Server is ready.
[2023-05-01 00:05:47 GMT+0400] Preloading IDs.
[2023-05-01 00:05:47 GMT+0400] Preloaded 5 IDs.```

**Comment by d-little** [comment]
> Checks fails but it successfully builds on my machine

They're just linting errors, run `npx eslint --cache src tests --ext ts --ext vue --fix`

**Comment by kberg** [comment]
Thanks for this. Please take a look at https://github.com/terraforming-mars/terraforming-mars/issues/4276, where someone was discussing a similar issue with Polish characters and the existing font. There were two other fonts also proposed.

Here was my comment:

> I don't think it's correct to completely replace the Prototype font. It's not known if this implementation of Prototype is compatible with the other languages. So I'd consider an implementation that only provides this version only when Polish is selected.

I realize that complicates things, but for now, let's isolate this change to translations, please.

**Comment by tltneon** [comment]
I propose that way to adding support for vary languages. If in needed you may easily add another font file with no extra modules or code rewrites.
image image

**Comment by kberg** [comment]
> I propose that way to adding support for vary languages. If it needed you may easily add another font file with no extra modules or code rewrites. image image UPD: added a few lines by that commit [aa81e3b](https://github.com/terraforming-mars/terraforming-mars/commit/aa81e3b7ba6b2084c51b87588dc993e2f5631ff8)

Yeah, that starts it. The hard part is getting the browser to ask for it.

Looks like Prototype is bound in CSS: 
image

and this might work, though I'm not certain: https://stackoverflow.com/questions/62349536/is-it-possible-to-dynamically-change-font-to-a-custom-typeface-with-javascript

File: src/client/components/card/CardContent.vue [hunk 1]
```diff
@@ -31,6 +32,9 @@ export default Vue.extend({
       type: Boolean,
       required: true,
     },
+    needSpacer: {
```
Comment (kberg) [inline src/client/components/card/CardContent.vue:35]: rename to `padBottom`?

File: src/styles/common.less [hunk 2]
```diff
@@ -1,244 +1,255 @@
...
+
+@font-face {
+    font-family: Prototype;
+    src: url(./assets/Prototype.ttf);
+    unicode-range: U+000-5FF; /* Latin glyphs */
+}
+@font-face {
+    font-family: Prototype;
+    src: url(./assets/PrototypeRussian.ttf);
+    unicode-range: U+0410-044F; /* Russian glyphs */
```
Comment (kberg) [inline src/styles/common.less:99]: OH THAT IS AWESOME.

File: src/styles/language_hacks.less [hunk 3]
```diff
@@ -1,353 +1,353 @@
...
+        }
+
+        &.card-asteroid-mining-consortium {
+            .card-requirements {
+                font-size: 14px;
+            }
+        }
+    }
+    .tag-none {
+        background-image: url(./assets/tags/tag-none-ru.png) !important;
```
Comment (kberg) [inline src/styles/language_hacks.less:127]: Nice!

File: src/styles/start_screen.less [hunk 4]
```diff
@@ -1,172 +1,172 @@
...
+    margin-bottom: 5px;
+    margin-left: 5px;
+    text-shadow: 0 0 2px black, 0 2px 2px black;
+    letter-spacing: 1px;
+    background-image: url("assets/buttons-homepage/planets.jpg");
+}
+
+.start-screen-title-top {
+    transform: scaleX(2);
+    text-align: center;
```
Comment (kberg) [inline src/styles/start_screen.less:85]: What is this change doing?

**Comment by kberg** [comment]
Oh wow there's some nice things in here now.

Since this PR is doing three things at once (adding translations, changing the font, and also changing rendering) and each of those are isolated, please can they be in different PRs? I can get the easy stuff out first.

Also, can you change the filenames of the new prototype fonts to have their two-character i18n langauge abbreviation (eg Prototype-ru, Prototype-pl)

**Comment by kberg** [comment]
(Sorry about the long delay. It's been quite a week. :D)

File: src/client/components/card/CardContent.vue (hunk 1 above)
Comment (tltneon) [reply src/client/components/card/CardContent.vue:35]: Got it

File: src/styles/start_screen.less (hunk 4 above)
Comment (tltneon) [reply src/styles/start_screen.less:85]: It sets heading exactly at the middle of screen. You can see mostly on other languages


image

Comment (kberg) [reply src/styles/start_screen.less:85]: OK that makes sense. Thank you.

**Comment by tltneon** [comment]
> Since this PR is doing three things at once (adding translations, changing the font, and also changing rendering) and each of those are isolated, please can they be in different PRs? I can get the easy stuff out first.

I'll split it out, no problem. 

> Also, can you change the filenames of the new prototype fonts to have their two-character i18n langauge abbreviation (eg Prototype-ru, Prototype-pl)

Okay.

> (Sorry about the long delay. It's been quite a week. :D)

I understand it, everybody has it own time where they busy :)

**Comment by kberg** [comment]
I think this can be closed! Nice work. Let me know if I'm mistaken about this PR.

**Pull request closed without merging**